Source/Core/Resource.js

```diff
@@ -1869,24 +1868,26 @@ define([
 
                 return Resource.fetchBlob({
                     url: url
-                });
-            })
-            .then(function(blob) {
-                if (!defined(blob)) {
-                    return;
-                }
+                })
+                .then(function(blob) {
+                    if (!defined(blob)) {
+                        deferred.reject('Failed to fetch blob.');
+                        return;
+                    }
 
-                return Resource.createImageBitmapFromBlob(blob, {
-                    flipY: flipY,
-                    premultiplyAlpha: false
-                });
-            })
-            .then(function(imageBitmap) {
-                if (!defined(imageBitmap)) {
-                    return;
-                }
+                    return Resource.createImageBitmapFromBlob(blob, {
+                        flipY: flipY,
+                        premultiplyAlpha: false
+                    });
+                })
+                .then(function(imageBitmap) {
+                    if (!defined(imageBitmap)) {
+                        deferred.reject('Failed to create imageBitmap from blob.');
```

Comment:
Same comment about error specificity and type but I'll also throw in a question: But does `Resource.createImageBitmapFromBlob` return undefined instead of failing itself?  Shouldn't it be the one generated the error?